misc/ls-extensions/modules/project-design-service/src/main/java/io/ballerina/projectdesign/ComponentModelBuilder.java

```diff
@@ -47,16 +48,20 @@ public ComponentModel constructComponentModel(Package currentPackage) {
         PackageId packageId = new PackageId(currentPackage);
 
         currentPackage.modules().forEach(module -> {
+            String moduleRootPath = module.project().sourceRoot().toAbsolutePath().toString();
+            if (module.moduleName().moduleNamePart() != null) {
+                moduleRootPath = moduleRootPath + File.separator + module.moduleName().moduleNamePart();
```

Comment:
What if we have subdirectories inside the module? Shall we verify whether this logic work in such scenarios as well?

Reply:
It seems that it is incorrect have sub-directories containing ballerina files inside a module. All the ballerina files should be located in the root level. @azinneera Could you please confirm ?

Reply:
> It seems that it is incorrect have sub-directories containing ballerina files inside a module. All the ballerina files should be located in the root level. @azinneera Could you please confirm ?

Correct.